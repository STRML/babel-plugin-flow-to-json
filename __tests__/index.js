const path = require('path');
const fs = require('fs');
const babel = require('babel-core');
const flowToJson = require('../');

const testsPath = path.join(__dirname, './code');

const transform = code =>
    babel.transform(code, {
        babelrc: false,
        presets: ['env'],
        plugins: ['transform-class-properties', flowToJson, 'transform-flow-strip-types']
    }).code;

const testFile = file => {
    const text = fs.readFileSync(`${testsPath}/${file}`).toString();
    expect(transform(text)).toMatchSnapshot();
};

const files = fs.readdirSync(testsPath);
files.forEach(file => {
    test(file, () => {
        testFile(file);
    });
});
