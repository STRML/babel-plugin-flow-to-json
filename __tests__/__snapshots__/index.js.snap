// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`object-array.js 1`] = `
"\\"use strict\\";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

var Bar = function Bar() {
    _classCallCheck(this, Bar);
};

Bar.prop = {
    type: \\"object\\",
    properties: {
        baz: {
            type: \\"array\\",
            items: {
                type: \\"object\\",
                properties: {
                    foo: {
                        type: \\"string\\"
                    },
                    bar: {
                        type: \\"number\\"
                    }
                },
                required: [\\"foo\\", \\"bar\\"]
            }
        },
        qux: {
            type: \\"object\\",
            properties: {
                foo: {
                    type: \\"string\\"
                },
                bar: {
                    type: \\"number\\"
                }
            },
            required: [\\"foo\\", \\"bar\\"]
        },
        quux: {
            type: \\"array\\",
            items: {
                type: \\"object\\",
                properties: {
                    foo: {
                        type: \\"string\\"
                    },
                    bar: {
                        type: \\"number\\"
                    }
                },
                required: [\\"foo\\", \\"bar\\"]
            }
        },
        foo: {
            type: \\"object\\",
            properties: {
                bar: {
                    type: \\"array\\",
                    items: {
                        type: \\"object\\",
                        properties: {
                            foo: {
                                type: \\"string\\"
                            },
                            bar: {
                                type: \\"number\\"
                            }
                        },
                        required: [\\"foo\\", \\"bar\\"]
                    }
                }
            },
            required: [\\"bar\\"]
        }
    },
    required: [\\"baz\\", \\"qux\\", \\"quux\\", \\"foo\\"]
};"
`;

exports[`opaque.js 1`] = `
"\\"use strict\\";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

var Foo = function Foo() {
    _classCallCheck(this, Foo);
};

Foo.props = {
    type: \\"object\\",
    properties: {
        id1: {
            type: \\"number\\"
        },
        id2: {
            type: \\"number\\"
        }
    },
    required: [\\"id1\\", \\"id2\\"]
};"
`;

exports[`optional-objects.js 1`] = `
"\\"use strict\\";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

var Bar = function Bar() {
    _classCallCheck(this, Bar);
};

Bar.prop = {
    type: \\"object\\",
    properties: {
        bar: {
            anyOf: [{
                type: \\"null\\"
            }, {
                type: \\"object\\",
                properties: {
                    foo: {
                        type: \\"string\\"
                    }
                },
                required: [\\"foo\\"]
            }]
        },
        baz: {
            anyOf: [{
                type: \\"null\\"
            }, {
                type: \\"array\\",
                items: {
                    type: \\"string\\"
                }
            }]
        },
        qux: {
            anyOf: [{
                type: \\"null\\"
            }, {
                type: \\"object\\",
                properties: {
                    foo: {
                        type: \\"string\\"
                    },
                    bar: {
                        type: \\"number\\"
                    }
                },
                required: [\\"foo\\", \\"bar\\"]
            }]
        },
        quux: {
            type: \\"object\\",
            properties: {
                foo: {
                    type: \\"number\\"
                }
            },
            required: [\\"foo\\"]
        }
    },
    required: [\\"bar\\", \\"baz\\", \\"qux\\", \\"quux\\"]
};"
`;

exports[`optional-primitives.js 1`] = `
"\\"use strict\\";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

var Bar = function Bar() {
    _classCallCheck(this, Bar);
};

Bar.prop = {
    type: \\"object\\",
    properties: {
        bar: {
            anyOf: [{
                type: \\"null\\"
            }, {
                type: \\"string\\"
            }]
        },
        baz: {
            anyOf: [{
                type: \\"null\\"
            }, {
                type: \\"number\\"
            }]
        },
        qux: {
            anyOf: [{
                type: \\"null\\"
            }, {
                type: \\"boolean\\"
            }]
        }
    },
    required: [\\"bar\\", \\"baz\\", \\"qux\\"]
};"
`;

exports[`primitives.js 1`] = `
"\\"use strict\\";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

var Foo = function Foo() {
    _classCallCheck(this, Foo);
};

Foo.testPrimitives = {
    type: \\"object\\",
    properties: {
        a: {
            type: \\"number\\"
        },
        b: {
            type: \\"string\\"
        },
        c: {
            type: \\"boolean\\"
        }
    },
    required: [\\"a\\", \\"b\\", \\"c\\"]
};"
`;

exports[`simple-union.js 1`] = `
"\\"use strict\\";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

var Bar = function Bar() {
    _classCallCheck(this, Bar);
};

Bar.prop = {
    anyOf: [{
        type: \\"string\\"
    }, {
        type: \\"object\\",
        properties: {
            foo: {
                type: \\"string\\"
            },
            bar: {
                type: \\"number\\"
            }
        },
        required: [\\"foo\\", \\"bar\\"]
    }, {
        type: \\"null\\"
    }]
};"
`;

exports[`type-alias.js 1`] = `
"\\"use strict\\";

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

var Foo = function Foo() {
    _classCallCheck(this, Foo);
};

Foo.baz = {
    type: \\"object\\",
    properties: {
        foo: {
            type: \\"string\\"
        }
    },
    required: [\\"foo\\"]
};
Foo.bar = {
    type: \\"array\\",
    items: {
        type: \\"number\\"
    }
};"
`;

exports[`typeof.js 1`] = `
"'use strict';

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

var user = {
    id: 'some id',
    name: 'some name'
};

var Foo = function Foo() {
    _classCallCheck(this, Foo);
};

Foo.prop = {
    type: 'object',
    properties: {
        user: {
            type: 'object',
            properties: {
                id: {
                    type: 'string'
                },
                name: {
                    type: 'string'
                }
            },
            required: ['id', 'name']
        }
    },
    required: ['user']
};"
`;

exports[`union-with-enums.js 1`] = `
"'use strict';

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\\"Cannot call a class as a function\\"); } }

var Bar = function Bar() {
    _classCallCheck(this, Bar);
};

Bar.prop = {
    enum: ['someStr', 46],
    anyOf: [{
        type: 'object',
        properties: {
            foo: {
                type: 'string'
            },
            bar: {
                type: 'number'
            }
        },
        required: ['foo', 'bar']
    }, {
        type: 'null'
    }]
};"
`;
